### PR TITLE
docs: react-tinacms-editor for developers

### PR DESCRIPTION
**Pull request by jpuri**
WYSIWYG Documentation
 https://github.com/tinacms/tinacms/pull/1310/files?short_path=ca19193#diff-ca191934cd6c29d12ba28b53ebbc917a

### CODE REVIEW

**Comment by jpuri** [comment]
Hey @ncphillips , @DirtyF : I have added more content to documentation. Plz share your feedbacks about it.

**Comment by DirtyF** [comment]
@logan-anderson for now we think it's already a good start to document packages in Markdown files. If all packages are documented then we can always envision listing them on the website à la Gatsby:  https://www.gatsbyjs.org/packages/gatsby-source-filesystem/?=

**Comment by ncphillips** [comment]
Yeah these docs aren't really for people using `react-tinacms-editor`, but for people looking to contribute to the package. 

I had a quick read and it looks great! I haven't had a chance to read deeply yet though. Hopefully I will soon.

**Comment by kendallstrautman** [comment]
LGTM!

File: packages/react-tinacms-editor/editor-docs/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  - MyPopup,...
+- nodeView
+- plugin
+- commands
+
+`Menu` folder has 2 components `MarkdownMenu` and `ProsemirrorMenu`, if present these are added to menubar in Markdown mode and Prosemirror mode respectively.
+
+`Popups` folder has various Popup components. For instance image and link have Popup components used for editing. These components should take care for show / hide themselves conditionally.
+
+`nodeView` is used to create customized vide for prosemirror node, this is used for code block and image nodes.
```
Comment (ncphillips) [inline packages/react-tinacms-editor/editor-docs/README.md:55]:
```suggestion
`nodeView` is used to create a custom view for the prosemirror node. This is used for code block and image nodes.
```

File: packages/react-tinacms-editor/editor-docs/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+`plugin` is prosemirror [plugin](https://prosemirror.net/docs/ref/#state.Plugin_System) object. It helps maintain state for plugin for instance for links and images it helps to keep information when user is currently editing link or image. It also enables other functionality like attaching event listeners, creation editor decorations, etc.
```
Comment (ncphillips) [inline packages/react-tinacms-editor/editor-docs/README.md:57]:
```suggestion
`plugin` is prosemirror [plugin](https://prosemirror.net/docs/ref/#state.Plugin_System) object. It helps maintain state for plugins, such as whether or not the user is currently editing a link or image. The `plugin` also enables other functionality like attaching event listeners, creation editor decorations, etc.
```

File: packages/react-tinacms-editor/editor-docs/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+`commands` keeping commands related to the plugin together.
```
Comment (ncphillips) [inline packages/react-tinacms-editor/editor-docs/README.md:59]:
```suggestion
`commands` defines the [prosemirror commands](https://prosemirror.net/docs/ref/#commands) related to the plugin.
```

File: packages/react-tinacms-editor/editor-docs/README.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+Currently all the `keyboard shortcuts` and prosemirror `input rules` are combined in `common plugin` it is required to refactor these into their respective plugins.
```
Comment (ncphillips) [inline packages/react-tinacms-editor/editor-docs/README.md:61]:
```suggestion
Currently all the [keyboard shortcuts](https://prosemirror.net/docs/ref/#keymap) and [input rules](https://prosemirror.net/docs/ref/#inputrules) are combined in the `common plugin`. Future work will move these into their respective plugins.
```

File: packages/react-tinacms-editor/editor-docs/README.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+### Markdown Editor
+
+[This](https://github.com/tinacms/tinacms/tree/master/packages/react-tinacms-editor/src/components/MarkdownEditor) editor is simple textarea where editor content can be edited a plain text. Currently menubar options are not enabled in this editor.
```
Comment (ncphillips) [inline packages/react-tinacms-editor/editor-docs/README.md:65]:
```suggestion
The [Markdown Editor](https://github.com/tinacms/tinacms/tree/master/packages/react-tinacms-editor/src/components/MarkdownEditor) is simple textarea where editor content can be edited a plain text. Currently menubar options are not enabled in this editor.
```

**Pull request merged**